test/shared/src/main/scala/zio/test/package.scala

```diff
@@ -109,7 +111,7 @@ package object test extends CompileVariants {
   /**
    * Checks the assertion holds for the given value.
    */
-  final def assert[A](value: => A, assertion: Assertion[A]): TestResult =
+  final def assert[A](value: => A)(assertion: Assertion[A]): TestResult =
```

Comment:
Coming back to this, it's not part of the type inference issue, is it? I was just wondering if it made the API more usable.

Reply:
Actually it is related to the type inference issue. We need to curry `assert` to "lock in" in the type of `A` so that the compiler properly widens the right hand side if it is a subtype of the left hand side, for example `assert(List(1, 2, 3))(equalTo(List.empty))`. And then we need the implicit evidence to prevent it from widening the type to `Any` so nonsensical comparisons don't compile.

Reply:
Brilliant!